### ISSUE TITLE
[BUG] Singleton PopupService uses scoped IDispatcher

### ISSUE THREAD

**Issue by jenscski** [CommunityToolkit/Maui#2375]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

When adding code for validating service scopes, after updating to version 9.1.1 of .NET MAUI Community Toolkit our application is not starting as we have enabled validating service scopes.

Exception message is
```
Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: CommunityToolkit.Maui.Core.IPopupService Lifetime: Singleton ImplementationType: CommunityToolkit.Maui.PopupService': Cannot consume scoped service 'Microsoft.Maui.Dispatching.IDispatcher' from singleton 'CommunityToolkit.Maui.Core.IPopupService'.)
``` 

### Expected Behavior

That the code works as before updating to version 9.1.1

### Steps To Reproduce

1. Download reproduction
2. Run .NET Maui app in reproduction solution

### Link to public reproduction project repository

https://github.com/jenscski/CommunityToolkitMaui2375

### Environment

```markdown
- .NET MAUI CommunityToolkit: 9.1.1
- OS:
- .NET MAUI: 8.0.x
```

### Anything else?

Was introduced by #1688

**Comment by jenscski** [comment]
Reproduction added

**Comment by jenscski** [comment]
This was actually fixed in #2333, so this is a duplicate issue.

But, after looking at the fix, it calls IDispatcherProvider.GetForCurrentThread in the constructor. This is a singleton, so the constructor may be called on a thread that is not the UI thread, or if it's a multi-window application, isn't there multiple UI threads?

Could that cause any problems?
